### PR TITLE
SDK integration

### PR DESCRIPTION
**Pull request by hrishi1096**

### CODE REVIEW

File: src/test/resources/conf/capabilities/browserstack-parallel-devices.yml [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+
+app: ./apps/browserstack-demoapp.apk
+
+# =======================================
+# Platforms (Browsers / Devices to test)
+# =======================================
+# Platforms object contains all the browser / device combinations you want to test on.
+# Entire list available here -> (https://www.browserstack.com/list-of-browsers-and-platforms/automate)
+
+platforms:
```
Comment (abhi-singhs) [inline src/test/resources/conf/capabilities/browserstack-parallel-devices.yml:34]: You'll need to add 2-3 devices to this profile.

**Review by abhi-singhs — CHANGES_REQUESTED** [review]
Do we need [LocalSample.apk](https://github.com/BrowserStackCE/browserstack-examples-appium-testng/blob/sdk_review/apps/LocalSample.apk) App?

**Comment by tschoem** [comment]
> Do we need [LocalSample.apk](https://github.com/BrowserStackCE/browserstack-examples-appium-testng/blob/sdk_review/apps/LocalSample.apk) App?

We've decided to add them even in this repo @abhi2810 : https://github.com/browserstack/testng-appium-app-browserstack/pull/24/files

because it is very unintuitive for customers using the repos without access to the apps...

**Comment by abhi-singhs** [comment]
> > Do we need [LocalSample.apk](https://github.com/BrowserStackCE/browserstack-examples-appium-testng/blob/sdk_review/apps/LocalSample.apk) App?
> 
> We've decided to add them even in this repo @abhi2810 : https://github.com/browserstack/testng-appium-app-browserstack/pull/24/files
> 
> because it is very unintuitive for customers using the repos without access to the apps...

I asked this question because this app is not used anywhere in the repo.

File: src/test/resources/conf/capabilities/browserstack-parallel-devices.yml (hunk 1 above)
Comment (tschoem) [reply src/test/resources/conf/capabilities/browserstack-parallel-devices.yml:34]: Done!

**Comment by tschoem** [comment]
> I asked this question because this app is not used anywhere in the repo.

Oh maybe it should be referenced in the readme / browserstack-local.yml files ?

**Comment by abhi-singhs** [comment]
> > I asked this question because this app is not used anywhere in the repo.
> 
> Oh maybe it should be referenced in the readme / browserstack-local.yml files ?

The LocalSample app is not relevant to this code repo. The main BrowserStack App has Local testing capabilities and the code is also written accordingly.

**Comment by tschoem** [comment]
ah ok i'll remove the changes and the app

**Comment by tschoem** [comment]
Some more notes here:
- the onprem configs don't run locally anymore. I'm not sure if we can change the pom file to fix this
- several of the tests actually fail. It might be worth fixing them for both sdk and regular branch

**Comment by hrishi1096** [comment]
LocalSample.apk has been removed already by Thomas.

As for the failing tests, I noticed that they are failing intermittently because the app crashes and this is when I run it with mvn. Strangely enough, when I run the same suite/profile with gradle, they all pass.

**Comment by hrishi1096** [comment]
There are some test failures and problems with the on-prem profile as stated in a previous comment but all the changes related to SDK conversion are done and tested.

Merging this PR now and will create a new ticket (and consequently a PR) to fix the other problems specifically.
Hope that's okay. @abhi2810 @tschoem

**Review by abhi-singhs — APPROVED** [review]

**Pull request merged**